### DISCUSSION TITLE
Getting the local time

### DISCUSSION THREAD

**Discussion by pnielan** [time-rs/time#547] [Question] [answered]
Following the example in the book introduction section:

1. I changed my cargo.toml file
2. I put `use time::OffsetDateTime` in my main.rs (and only file). 
3. I can call and use as follows `let now = OffsetDateTime::now_utc();` with no problem.
4. But when I change utc to local as in `let now = OffsetDateTime::now_local();` I get an compilation error.

```
|177 |     let now = OffsetDateTime::now_local();
|                               ^^^^^^^^^
|                               |
|                               function or associated item not found in OffsetDateTime 
|                               help: there is an associated function with a similar name: now_utc
```

What am I doing wrong? And thanks for your attention and patience.

**Comment by jhpratt** [comment, marked as answer]
If you look at [the documentation for the method](https://time-rs.github.io/api/time/struct.OffsetDateTime.html#method.now_local), you'll see that you have to enable the `local-offset` feature.

**Reply by pnielan** [reply]
Thanks very much. I looked at the documentation. I do not understand what "enable the local-offset feature" means in terms of changing my main.rs file.

**Reply by pnielan** [reply]
I looked and saw `Available on crate feature local-offset only.` but not sure what to do with that.

**Reply by jhpratt** [reply]
I take it you're new to Rust. My recommendation is to read [this page](https://doc.rust-lang.org/cargo/reference/features.html#dependency-features) from the Cargo book.

**Reply by pnielan** [reply]
Thanks for the clue. I studied features on other web page for other packages.

Look like I had to put this in the cargo.toml file. I am a long-time programmer, but a new RUST programmer and am grasping the lingo slowly.

```
[dependencies]
time = { version = "0.3", features = ["macros", "local-offset"]}
```

**Reply by jhpratt** [reply]
It can definitely be overwhelming when you're just starting. That line is correct, though!

**Reply by pnielan** [reply]
To close this off, in case it help someone else, I looked at 

`https://stackoverflow.com/questions/58480205/how-do-you-enable-a-rust-crate-feature`

to clue me in. Good breakthrough, I am looking forward to starting to add and use more crates.io items.
